dash.el

```diff
@@ -2274,9 +2274,22 @@ The test for equality is done with `equal',
 or with `-compare-fn' if that's non-nil.
 
 Alias: `-uniq'"
-  (let (result)
-    (--each list (unless (-contains? result it) (!cons it result)))
-    (nreverse result)))
+  ;; Implementation note: Using a hash-table to remove duplicates is
+  ;; considerably faster than do it by iteration when the length
+  ;; of the list is greater than 32.
+  (let* ((-compare-fn (or -compare-fn #'equal))
+         (len (length list))
+         (lut (and (> len 32)
+                   (memq -compare-fn '(equal eq eql))
+                   (make-hash-table :test -compare-fn))))
+    (if lut
+        (--remove (when (gethash it lut)
+                    (puthash it t lut))
```

Comment:
This can't be right: when `gethash` returns `nil`, the `puthash` won't be executed. The correct logic is the following:

```el
(--filter (unless (gethash it ht)
            (puthash it t ht))
          list)
```
```